Sources/Core/Shared/RepeatedOperation.swift

```diff
@@ -299,6 +299,7 @@ public class RepeatedOperation<T where T: NSOperation>: GroupOperation {
      can prevent another operation from being added.
     */
     public func addNextOperation(@autoclosure shouldAddNext: () -> Bool = true) -> Bool {
+        guard !cancelled else { return false }
         guard shouldAddNext(), let payload = next() else { return false }
```

Comment:
I think maybe it's nicer to add the `!cancelled` to the guard that is already there?

```swift
guard !cancelled && shouldAddNext(), let payload = next() else { return false }
```